inst/unitTests/runit.Matrix.R

```diff
@@ -171,4 +171,16 @@ if (.runThisTest) {
       checkEquals( m[,1], m[,2] )
     }
 
+    test.IntegerMatrix.accessor.with.bounds.checking <- function() {
+        m <- matrix(seq(1L, 12, by=1L), nrow=4L, ncol=3L)
+        checkEquals(mat_access_with_bounds_checking(m, 0, 0), 1)
+        checkEquals(mat_access_with_bounds_checking(m, 1, 2), 10)
+        checkEquals(mat_access_with_bounds_checking(m, 3, 2), 12)
+        checkException(mat_access_with_bounds_checking(m, 4, 2) , msg = "index out of bounds not detected" )
```

Comment:
Am I missing something or should the msg be "index out of bounds" because the fact that we have an exception means your `throw` went into effect?